### PR TITLE
feat: plot matching sprite func

### PR DESCRIPTION
**Pull request by makram93**
This PR adds functionality to show matching images in a sprite. The leftmost is the query followed by matching results. There is a gap between the query and its matches to distinguish/differentiate.
image

### CODE REVIEW

File: docarray/document/mixins/plot.py [hunk 1]
```diff
@@ -97,6 +102,110 @@ def display(self):
...
+        if not self.uri and self.tensor is None:
+            raise ValueError(
+                f'Document has neither `uri` nor `tensor`, cannot be plotted'
+            )
+
+        if top_k <= 0:
+            raise ValueError(f'`limit` must be larger than 0, receiving {top_k}')
+
+        if image_source not in ('uri', 'tensor'):
+            raise ValueError(f'image_source can be only `uri` or `tensor`')
```
Comment (bwanglzu) [inline docarray/document/mixins/plot.py:138]: with the literal check, this is not needed anymore

File: docarray/document/mixins/plot.py [hunk 2]
```diff
@@ -97,6 +102,110 @@ def display(self):
...
+        img_size = int((canvas_size - 50) / img_per_row)
+        if img_size < min_size:
+            # image is too small, recompute the image size and canvas size
+            img_size = min_size
+            canvas_size = img_per_row * img_size + 50
+
+        _d = copy.deepcopy(self)
+        if image_source == 'uri':
+            _d.load_uri_to_image_tensor()
+            channel_axis = -1
```
Comment (bwanglzu) [inline docarray/document/mixins/plot.py:155]: maybe you need `load_uri_to_image_tensor(channel_axis=channel_axis)`? then why do you need `channel_axis` as an argument pass into this function?

**Review by bwanglzu — COMMENTED** [review]
left some comments

File: docarray/document/mixins/plot.py [hunk 3]
```diff
@@ -97,6 +102,110 @@ def display(self):
...
     plot = deprecate_by(display, removed_at='0.5')
 
+    def plot_matching_sprites(
+        self,
+        output: Optional[str] = None,
+        canvas_size: int = 1920,
+        channel_axis: int = -1,
+        top_k: int = 5,
+        min_size: int = 100,
+        image_source: str = 'tensor',
```
[the remark this answers is not in the document]
Comment (makram93) [reply docarray/document/mixins/plot.py]: Thanks for the pointer :)

**Comment by bwanglzu** [comment]
@makram93 mybad, `literal` only works on py3.8+, please ignore my first 2 comments.

File: docarray/document/mixins/plot.py [hunk 4]
```diff
@@ -97,6 +102,103 @@ def display(self):
 
     plot = deprecate_by(display, removed_at='0.5')
 
+    def plot_matching_sprites(
```
Comment (hanxiao) [inline docarray/document/mixins/plot.py:105]:
```suggestion
    def plot_matches_sprites(
```

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Review by hanxiao — CHANGES_REQUESTED** [review]
need a doc, subsection under this https://docarray.jina.ai/fundamentals/documentarray/visualization/#image-sprites

**Review by hanxiao — APPROVED** [review]

**Pull request merged**